### PR TITLE
update numpy easyblock to be able to use FlexiBLAS with MKL as dependency

### PR DESCRIPTION
**Pull request by SebastianAchilles**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by surak** [comment]
It still asks for mkl on line 150

File: easybuild/easyblocks/n/numpy.py [hunk 1]
```diff
@@ -79,7 +79,8 @@ def configure_step(self):
             "search_static_first=True",
         ])
 
-        if get_software_root("imkl"):
+        # If FlexiBLAS has a dependcy to MLK we want to link to FlexiBLAS and not to MKL.
```
Comment (ocaisa) [inline easybuild/easyblocks/n/numpy.py:82]:
```suggestion
        # If FlexiBLAS has a dependency on MKL we want to link to FlexiBLAS and not to MKL.
```

**Comment by boegel** [comment]
@surak Should be fixed with [a97f70e](https://github.com/easybuilders/easybuild-easyblocks/pull/2999/commits/a97f70e0c5c9a0106c225e2d6cc0630fc1ae155b), please try again with `--include-easyblocks-from-pr 2999` ?

**Comment by surak** [comment]
@boegel Works for me now!

File: easybuild/easyblocks/n/numpy.py [hunk 2]
```diff
@@ -79,7 +79,10 @@ def configure_step(self):
             "search_static_first=True",
         ])
 
-        if get_software_root("imkl"):
+        # If FlexiBLAS has a dependency on MKL we want to link to FlexiBLAS and not to MKL.
+        imkl_direct = get_software_root("imkl") and not get_software_root("FlexiBLAS")
```
Comment (akesandgren) [inline easybuild/easyblocks/n/numpy.py:83]: The comment and the code doesn't really match up.
The code does not check if FlexiBLAS has a dependency on MKL.
All it does is check if FlexiBLAS is available and if so ignores MKL.

But that still ens up doing what it should, so the comment should be changed to something more relevant.

**Comment by SebastianAchilles** [comment]
Thanks for spotting the problem @surak and thanks to @ocaisa and @boegel for improving the PR :+1:

File: easybuild/easyblocks/n/numpy.py (hunk 2 above)
Comment (SebastianAchilles) [reply easybuild/easyblocks/n/numpy.py:83]: Do you know if there is a way to check the dependencies of an EasyConfig?

Right, in the comment I described when/why this check is needed, but not what the code is doing. Let me try to improve the comment.

Comment (SebastianAchilles) [reply easybuild/easyblocks/n/numpy.py:83]: @akesandgren Do you think this improves the comment?

```suggestion
        # If both FlexiBLAS and MKL are found, we assume that FlexiBLAS has a dependency on MKL.
        # In this case we want to link to FlexiBLAS and not directly to MKL.
        imkl_direct = get_software_root("imkl") and not get_software_root("FlexiBLAS")
```

Comment (akesandgren) [reply easybuild/easyblocks/n/numpy.py:83]: At least it is more in line with what the code does.

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _SciPy-bundle-2023.07-gfbf-2023a.eb_ 
 * **SUCCESS** _SciPy-bundle-2021.10-intel-2021b.eb_ 

Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3171.skitty.os - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/boegel/3e81ac1000fdad63514a3800bd52a9f7 for a full test report.

**Pull request merged**